fix startup

Deutschi opened the pull request:
change start behaviour, change selector to add the trelabel button

Commit:
fix startup

change start behaviour, change selector to add the trelabel button

fredericseiler commented on trelabels.js:
```diff
@@ -68,13 +68,13 @@ function addTrelabelsButton() {
     return;
   }
 
-  if ($('.header-boards-button').length === 0) {
```
`.header-boards-button` still exists.

![image](https://user-images.githubusercontent.com/954168/57438929-3b9d8680-7245-11e9-94b1-097b8b3ef548.png)

Deutschi replied:
This is really odd, in my browser I have this:

![Anmerkung 2019-05-09 105419](https://user-images.githubusercontent.com/2011264/57440831-03984280-7249-11e9-8f25-87cd78801e4a.jpg)

Therefore I chose a different approach in that line.

fredericseiler replied:
Can you check with the Chrome Console if `$('#header').find('input').parent()` gives you the same element as `input[type="search"]` ?

The input in the "old" version of Trello is a `type="text"`...

Deutschi replied:
Yep, that works.

fredericseiler replied:
I can blindly push the changes and hope for the best, but from what I see on your screenshot, all class names are now generated. So the button and menu, which uses the old class names, will still not show up correctly...

Deutschi replied:
I will do another test round and try to test it in both variants.

Deutschi replied:
`$('#header').find('input').parent().prepend(button);` looks good in Chrome ("new") and Firefox ("old"). The Trelabel button is there and functional. However, I don't have any other extension for Trello, so I hope this will not interfere.

fredericseiler commented on trelabels.js:
```diff
@@ -214,6 +214,6 @@ function setStyle(newStyle) {
 }
 
-$(function () {
+$(window).on('load', function() {
```
We just need the DOM to be loaded, can you give me more details about this change ?

Deutschi replied:
I had this error:

![Anmerkung 2019-05-09 105304](https://user-images.githubusercontent.com/2011264/57440962-435f2a00-7249-11e9-9b4c-df06c92ae8ad.jpg)

The function keepTrelabelsButton() was executed to soon, so my goal was to call it later when everything is really loaded.

fredericseiler replied:
Executed after the DOM is loaded but before `#header` is a node, that's also weird...

fredericseiler replied:
It's time for another guess ! In the new version of Trello, is `#header` added to the DOM via JS ?

fredericseiler replied:
Ok, my bad, even in the old version, it's added via JS.

Deutschi replied:
The new version uses react, so I would say "yes" without taking a deeper look.

fredericseiler replied:
A better way to handle this is to loop until `#header` appears, like what's done in [`addTrelabelsButton()`](https://github.com/fredericseiler/trelabels/blob/master/trelabels.js#L71-L75).

fredericseiler replied:
What's your `window.trelloVersion` result in the Chrome Console ? build-3124 ?

Deutschi replied:
Yes, "build-3124". A loop would also work and might be more robust.

Deutschi replied:
How about just moving the `#header` loop from `addTrelabelsButton()` to `init` and call that instead? Like:

```
function init() {
	
  if ($('#header').length === 0) {
    setTimeout(init, 1000);
    return;
  }
	
  $('head').append(font);
  addTrelabelsButton();
  keepTrelabelsButton();
  listenForClick();
  listenForResize();
}
```

fredericseiler replied:
Yep, makes more sense to delay the whole `init`.

So, we have the same Trello build but not the same content. Can you give it a try in incognito mode, without any other extension enabled ?

Deutschi replied:
I tried it five times in incognito and always got the react version.

fredericseiler replied:
Side node : the "old" version is also using React.

Deutschi replied:
Ah ok, I didn't notice :)

I am not really familiar with Github... how can I update the pull request with the changes we discussed? Just a commit in my fork to the master branch?

fredericseiler replied:
Yes, the PR will auto-update when you push to the same branch (master) in your fork.

You can also [allow me](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to do the changes on your PR, if you prefer.

Commit:
Update trelabels.js

1) revert the load function
2) start functions within init() when #header is present
3) change selector for search box so it works in every version of Trello

Deutschi commented:
I created a new commit. Is this version working for you as well?

fredericseiler commented:
Thanks (as always) Deutschi ! 👍
